src/Sylius/Component/Core/OrderProcessing/OrderTaxesProcessor.php

```diff
@@ -101,4 +107,13 @@ private function clearTaxes(OrderInterface $order): void
             $shipment->removeAdjustments(AdjustmentInterface::TAX_ADJUSTMENT);
         }
     }
+
+    private function getTaxAddress(OrderInterface $order): ?AddressInterface
+    {
+        if (!$this->defaultTaxationStrategy) {
+            return $order->getShippingAddress();
+        }
+
+        return $order->getBillingAddress();
+    }
```

Comment:
What about extracting this code to a separate service? It would provide even more flexibility - for example, one may adjust returned logic based on a current channel.